### ISSUE TITLE
babe: clarify `UnexpectedEpochChange`

### ISSUE THREAD

**Issue by liuchengxu** [paritytech/polkadot-sdk#2815]
If I'm not wrong, the infamous `unexpected epoch change` error so far was all caused by having no blocks produced within a babe session, which is no longer an issue since https://github.com/paritytech/substrate/pull/11727. Is it okay to remove it from now on? Or is there any other exception that may run into this error? cc @andresilva 

https://github.com/paritytech/polkadot-sdk/blob/7070b65d76ba7e99ea17b5506b4f1aa1d7545e4f/substrate/client/consensus/babe/src/lib.rs#L1520-L1522

**Comment by davxy** [comment]
The error is returned if we find an epoch change digest entry in the header when it is not expected to be present (e.g. in the mid of the epoch). Thus the error is not related to the problem solved by https://github.com/paritytech/substrate/pull/11727 

Feel free to re-open if this doesn't answer to your question

**Comment by liuchengxu** [comment]
That makes sense to me.

We recently encountered the `unexpected epoch change` problem on a live network due to no one being able to produce any blocks in a session, the chain was then bricked. It was using an older version of substrate that didn't include patch 11727, and the time warp hack didn't work out, so I resorted to 11727 with this error temporarily disabled. The chain is now revived, I'm asking as I'm not sure we could disable it permanently. Now I think we should get it back. Let me know if you have any advices for our situation and thank you for the answer! @davxy
